### ISSUE TITLE
can't update

### ISSUE THREAD

**Issue by graingert** [SirVer/ultisnips#685]
Looks like there was a force update on master. This breaks bundle
```
[2016-04-13 14:51:37] Plugin SirVer/ultisnips
[2016-04-13 14:51:37] $ cd '/home/graingert/.vim/bundle/ultisnips' && git pull && git submodule update --init --recursive
[2016-04-13 14:51:37] > From https://github.com/SirVer/ultisnips
[2016-04-13 14:51:37] >  + 7861450...5d4094b master     -> origin/master  (forced update)
[2016-04-13 14:51:37] > Recorded preimage for 'test/test_Autotrigger.py'
[2016-04-13 14:51:37] > Auto-merging test/test_Autotrigger.py
[2016-04-13 14:51:37] > CONFLICT (content): Merge conflict in test/test_Autotrigger.py
[2016-04-13 14:51:37] > Automatic merge failed; fix conflicts and then commit the result.
[2016-04-13 14:51:37] > 
```

**Comment by fgsch** [comment]
Same for plugged. I fixed it doing `git checkout HEAD`.
Looks like the repo was rebased at some point.

**Comment by graingert** [comment]
yeah I can fix it myself, but this is not the Vundle way.

**Comment by seletskiy** [comment]
@graingert: There is nothing wrong with amending commits, so if Vundle can't handle it, it's Vundle issue, not project.

**Comment by graingert** [comment]
@seletskiy but your docs push people to Vundle.

**Comment by fgsch** [comment]
@seletskiy I couldn't disagree more. There is a lot of things wrong with rebasing a public repository.
That said, it's your project, so you can see as you see fit, just don't blame 3rd parties.

**Comment by seletskiy** [comment]
@fgsch: For the plugin manager it should not be the issue, because repository is simple source of code, not complicated dev tool. So, it _should_ forcefully pull latest changes.

@graingert: I was not aware of that shortcoming of Vundle.

**Comment by fgsch** [comment]
Plugged is on the same camp, and probably others too. Fact is, most tools don't expect rebases and there is little to no sense doing so on a public repository.

**Comment by graingert** [comment]
@seletskiy it's handy because my own changes get merged in nicely

**Comment by graingert** [comment]
@seletskiy there's also this option in GitHub to protect master https://help.github.com/articles/configuring-protected-branches/

**Comment by seletskiy** [comment]
@graingert: If you have your own changes, that are not committed/pushed, which get rebased on top of the origin master, then you're not using git right. BTW, if that changes are useful, why not to create PR?

And again, there's nothing wrong with pushing with force, and Vundle should care about it. So I suppose to create issue in the Vundle repo and see what author thinking about it.

**Comment by dkarter** [comment]
@seletskiy I also encountered an issue with Vim-Plug. 

I agree with you that it's ok to push force, but only on feature branches (and not historical commits before the branch stem). 

Master branch should never be force pushed - that's generally considered a really bad practice and there are hundreds of blog posts, articles etc of folks explaining why. 

Considering that it actually affects how most Vim plugin managers handle updates this should really be avoided at all costs in future updates.

Sorry if I came off harsh, that's not my intention and I do appreciate your hard work.
